src/particle/particle/particle.py

```diff
@@ -285,6 +285,8 @@ def to_list(
         exclude_fields=(),  # type: Iterable[str]
         n_rows=-1,  # type: int
         filter_fn=None,  # type: Optional[Callable[[Particle], bool]]
+        particle=None,  # type: Optional[bool]
+        **search_terms  # type: Any
     ):
         # type: (...) -> List[List[Any]]
```

Comment:
It would be nice to have something more concrete here eventually. Like `List[Tuple[float, float, str, blah, blah]]`. Assuming the particle table has a strong structure to it?

Reply:
Yes. I would not know how mypy would say "potentially a list of arguments that can be bool, int, str, float" (probably that's it).  Can we leave this for now unless you know how to express the beast?

Reply:
Ahh, but it can change with the arguments. Okay, this is likely the best that we can do.

Reply:
Yes, a `ResType = Union[bool, int, str, float]` would be better than `Any`, but sadly you can't describe the ordering, since it depends on the arguments. But `List[List[ResType]` would be better (or just write it inline instead of making an alias).

Reply:
You lost me here. Should I then replace `Any` with `List[List[Union[bool, int, str, float]]]`?

Reply:
```suggestion
        # type: (...) -> List[List[Union[bool, int, str, float]]]
```
This is the simplest thing, though you could stick `Something = Union[bool, int, str, float]` somewhere in the file then use that instead of nesting it. But this is fine.

Reply:
OK, thanks.

Reply:
For reference, this caused trouble. I reverted back but will put it on my to-do list to look better at the issue, which comes from inside `to_dict` when calling `to_list` internally.

Reply:
I fixed it but you merged already. The problem was that you claimed this returned a dist with `List[str]` as a key, which is invalid, you can't have a mutable item as a key. Fixing the return type fixed the issue (plus a little cleanup).

Reply:
My apologies for that @henryiii.